### PR TITLE
kata-deploy: Add a new config file and runtimeClass for using Kata Containers with nydus

### PR DESCRIPTION
**Pull request by fidencio**
Let's allow users to easily do tests with both virtiofsd and nydusd without having to change Kata Containers configurations.

This PR is yet in draft stage as I didn't have time to test it with kata-deploy.

### CODE REVIEW

File: src/runtime/Makefile [hunk 1]
```diff
@@ -134,6 +135,8 @@ QEMUVALIDHYPERVISORPATHS := [\"$(QEMUPATH)\"]
 
 QEMUVIRTIOFSPATH := $(QEMUBINDIR)/$(QEMUVIRTIOFSCMD)
 
+QEMUNYDUSPATH := $(QEMUBINDIR)/$(QEMUNYDUSCMD)
```
Comment (liubin) [inline src/runtime/Makefile:138]: Is this `QEMUNYDUSPATH` needed? It is not used in this `Makefile` or the `configuration-qemu-nydus.toml.in`

File: src/runtime/Makefile [hunk 2]
```diff
@@ -354,6 +380,10 @@ ifeq ($(DEFAULT_HYPERVISOR),$(HYPERVISOR_QEMU_VIRTIOFS))
     DEFAULT_HYPERVISOR_CONFIG = $(CONFIG_FILE_QEMU_VIRTIOFS)
 endif
 
+ifeq ($(DEFAULT_HYPERVISOR),$(HYPERVISOR_QEMU_NYDUS))
```
Comment (liubin) [inline src/runtime/Makefile:383]: `HYPERVISOR_QEMU_NYDUS` is not defined. Should add to here?

https://github.com/kata-containers/kata-containers/blob/df3d9878d50a83d1576cdaf5701990e0827da019/src/runtime/Makefile#L121-L130

File: src/runtime/Makefile [hunk 3]
```diff
@@ -200,6 +203,15 @@ DEFVIRTIOFSQUEUESIZE ?= 1024
 # see `virtiofsd -h` for possible options.
 # Make sure you quote args.
 DEFVIRTIOFSEXTRAARGS ?= [\"--thread-pool-size=1\", \"-o\", \"announce_submounts\"]
+DEFSHAREDFS_QEMU_NYDUS := virtio-fs-nydus
+DEFNYDUSDAEMON := $(LIBEXECDIR)/nydusd
+DEFVALIDNYDUSDAEMONPATHS := [\"$(DEFNYDUSDAEMON)\"]
+# Default DAX mapping cache size in MiB
+#if value is 0, DAX is not enabled
```
Comment (studychao) [inline src/runtime/Makefile:210]: white space needed here? 
`#if value ...` -> `# if value ...`

**Comment by fidencio** [comment]
This can be dropped as `ConfigPath` actually works as expected.

**Pull request closed without merging**